sgkit/stats/ibs.py

```diff
@@ -0,0 +1,260 @@
+    """
+    ds = define_variable_if_absent(
+        ds,
+        variables.call_allele_frequency,
+        call_allele_frequency,
+        infer_call_allele_frequencies,
+    )
+    variables.validate(
+        ds, {call_allele_frequency: variables.call_allele_frequency_spec}
+    )
+
+    af = da.asarray(ds[call_allele_frequency])
+
+    ibs = da.blockwise(identity_by_state, "nxy", af, "nxa", af, "nya", concatenate=True)
+    new_ds = create_dataset(
+        {variables.ibs_probabilities: (("variants", "samples_0", "samples_1"), ibs)}
+    )
+    return conditional_merge_datasets(ds, new_ds, merge)
+
+
+def Weir_Goudet_beta(
+    ds: Dataset,
+    *,
+    ibs_probabilities: Hashable = variables.ibs_probabilities,
+    merge: bool = True,
+) -> Dataset:
+    """Estimate pairwise beta based on allelic matching as described in Weir
+    and Goudet 2017 [1].
+
+    Beta is the coancestry scaled by the average coancestry of all pairs of
+    individuals in the dataset such that the non-diagonal (non-self) values
+    sum to zero.
+
+    Parameters
+    ----------
+    ds
+        Genotype call dataset.
+    ibs_probabilities
+        Input variable name holding ibs_probabilities as defined by
+        :data:`sgkit.variables.ibs_probabilities_spec`.
+        If the variable is not present in ``ds``, it will be computed
+        using :func:`pairwise_ibs`.
+    merge
+        If True (the default), merge the input dataset and the computed
+        output variables into a single dataset, otherwise return only
+        the computed output variables.
+        See :ref:`dataset_merge` for more details.
+
+    Returns
+    -------
+    A dataset containing :data:`sgkit.variables.stat_Weir_Goudet_beta_spec`
+    of estimated pairwise coancestry relative to the average coancestry of all pairs of
+    individuals in the dataset.
+    The dimensions are named ``samples_0`` and ``samples_1``.
+
+    References
+    ----------
+    [1] - Bruce, S. Weir, and Jérôme Goudet 2017.
+    "A Unified Characterization of Population Structure and Relatedness."
+    Genetics 206 (4): 2085-2103.
+    """
+    ds = define_variable_if_absent(
+        ds, variables.ibs_probabilities, ibs_probabilities, pairwise_ibs
+    )
+    variables.validate(ds, {ibs_probabilities: variables.ibs_probabilities_spec})
+    ibs = da.nanmean((ds[ibs_probabilities]), axis=0)
+    x, y = ibs.shape
+    assert x == y
+    # mean of non-diagonal elements
+    avg = (da.sum(ibs) - da.sum(da.diagonal(ibs))) / (x * (y - 1))
+    beta = (ibs - avg) / (1 - avg)
+    new_ds = create_dataset(
+        {
+            variables.stat_Weir_Goudet_beta: (
+                ("samples_0", "samples_1"),
+                beta,
+            )
+        }
+    )
+    return conditional_merge_datasets(ds, new_ds, merge)
+
+
+def Weir_Goudet_beta_corrected(
+    ds: Dataset,
+    *,
+    beta0: float,
+    stat_Weir_Goudet_beta: Hashable = variables.stat_Weir_Goudet_beta,
+    merge: bool = True,
```

Comment:
Seems unnecessary to have a separate function for this - we could simply add the ``beta0`` parameter to ``Weir_Goudet_beta`` - if it's None, we do the uncorrected version and otherwise we do this.

Reply:
That was my first thought as well. The issue is that one usually needs to see the values of beta calculated from the data in order to pick a sensible value for `beta0`, which would require running the function twice. It's actually tempting to remove the correction function completely and just describe the process in the docstring of `Weir_Goudet_beta` with an example (it's quite simple).

Reply:
That does seem like a sensible approach. I don't think 
```python
x = func()
y = func(starting_point=x)
```
is any worse than 
```python
x = func()
y = func_with_starting_point(x)
```
though? Anyway, pulling the correction out of this PR seems like a good approach for the moment.